labrad/test/test_signal.py

```diff
@@ -0,0 +1,66 @@
+import Queue
+
+import pytest
+
+import labrad
+from labrad.server import LabradServer, setting, Signal
+import labrad.util
+
+
+class SignalTestServer(LabradServer):
+    name = "Signal Test Server"
+
+    class_signal = Signal(100, 'class_signal', '?')
+
+    def __init__(self):
+        super(SignalTestServer, self).__init__()
+        self.instance_signal = Signal(200, 'instance_signal', '?')
+
+    @setting(101)
+    def fire_class_signal(self, c, data):
+        self.class_signal(data)
+
+    @setting(201)
+    def fire_instance_signal(self, c, data):
+        self.instance_signal(data)
+
+
+def _test_signals(signal_setting, fire_setting):
+    with labrad.util.syncRunServer(SignalTestServer()):
+        with labrad.connect() as cxn:
+            server = cxn.signal_test_server
+            msg_id = 12345
+
+            # add a listener to enqueue all messages
+            queue = Queue.Queue()
+            def handler(message_ctx, msg):
+                queue.put((message_ctx, msg))
+            cxn._backend.cxn.addListener(handler, ID=msg_id)
+
+            # we don't get messages before signing up
+            server[fire_setting]('not listening')
```

Comment:
I am down with the idea, but haven't reviewed the code.  @jayich, feel free
to approve this.


On Wed, Apr 20, 2016 at 6:35 PM, jayich <notifications@github.com> wrote:

> In labrad/test/test_signal.py
> <https://github.com/labrad/pylabrad/pull/245#discussion_r60515041>:
>
> > +
> > +
> > +def _test_signals(signal_setting, fire_setting):
> > +    with labrad.util.syncRunServer(SignalTestServer()):
> > +        with labrad.connect() as cxn:
> > +            server = cxn.signal_test_server
> > +            msg_id = 12345
> > +
> > +            # add a listener to enqueue all messages
> > +            queue = Queue.Queue()
> > +            def handler(message_ctx, msg):
> > +                queue.put((message_ctx, msg))
> > +            cxn._backend.cxn.addListener(handler, ID=msg_id)
> > +
> > +            # we don't get messages before signing up
> > +            server[fire_setting]('not listening')
>
> @maffoo <https://github.com/maffoo> , thank you, this is good to know.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/labrad/pylabrad/pull/245/files/d2a880f9ae576924aaa3e174a9a19f4179c28201..e1dec58f41804f44f8c2cf60a3a51489041db509#r60515041>
>